### PR TITLE
FOGL-7103: Modified to restart notification service in case of SIGTERM…

### PR DESCRIPTION
**Pull request by gnandan**
… signal

Signed-off-by: nandan <nandan.ghildiyal@gmail.com>

### CODE REVIEW

**Review by MarkRiddoch — CHANGES_REQUESTED** [review]
I think we have a fundamental misunderstanding here. The expectation was that the process should be restarted by the core, not that the process calls an internal restart routine. Hence the suggested change was to handle SIGTERM as it was being handled, just avoid the step of unregistering with the core as that stopped the monitoring services within the Fledge core restarting the process.

File: C/services/common/include/notification_service.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ class NotificationService : public ServiceAuthHandler
 		const std::string&	getName() { return m_name; };
 		bool 			start(std::string& coreAddress,
 					      unsigned short corePort);
-		void 			stop();
+		void 			stop(bool remvoeFromCore=true);
```
Comment (MarkRiddoch) [inline C/services/common/include/notification_service.h:38]: typo removeFromCore

File: C/services/common/notification_service.cpp [hunk 2]
```diff
@@ -313,10 +322,14 @@ bool NotificationService::start(string& coreAddress,
  * Unregister notification subscriptions and
  * stop NotificationAPi listener
  */
-void NotificationService::stop()
+void NotificationService::stop(bool remvoeFromCore)
```
Comment (MarkRiddoch) [inline C/services/common/notification_service.cpp:325]: removeFromCore

File: C/services/common/notification_service.cpp [hunk 3]
```diff
@@ -313,10 +322,14 @@ bool NotificationService::start(string& coreAddress,
...
 {
 	m_logger->info("Stopping Notification service '" + m_name + "' ...");
 
+	if (remvoeFromCore == false)
```
Comment (MarkRiddoch) [inline C/services/common/notification_service.cpp:329]: removeFromCore

**Review by MarkRiddoch — CHANGES_REQUESTED** [review]
Just minor typo

File: C/services/common/notification_service.cpp (hunk 3 above)
Comment (gnandan) [reply C/services/common/notification_service.cpp]: typo is fixed

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**